layout/layers_layers_assigner.go

```diff
@@ -30,6 +30,10 @@ func maxNodeID(g Graph) uint64 {
 
 func assignLevels(g Graph) map[uint64][2]int {
 	nodeYX := make(map[uint64][2]int, len(g.Nodes))
+	neighbors := make(map[uint64][]uint64)
```

Comment:
pre-allocating at least to node size
```suggestion
	neighbors := make(map[uint64][]uint64, len(g.Nodes))
```